### PR TITLE
Fix centipede compilation

### PR DESCRIPTION
**Pull request by jonathanmetzman**
1. Don't use trace-loads as it is not fully supported in LLVM and breaks many projects (see https://github.com/google/centipede/commit/9383870160a8cb1d10ec15e9ca85d4b589c8151e)
2. Use -Wl for link time options so they aren't unnecessarily used (also fixes meson based projects).

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
We shouldn't have added any sanitizer flags to those centipede gave us. Lesson learned here.

**Comment by jonathanmetzman** [comment]
I think this fixes the last big batch of projects (~10 or so that use meson) that centipede doesn't work on

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py all --sanitizer none --fuzzing-engine centipede

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py all --sanitizer none --fuzzing-engine centipede

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py all --sanitizer none --fuzzing-engine centipede

**Comment by jonathanmetzman** [comment]
Related: https://github.com/google/oss-fuzz/issues/9299 https://github.com/google/oss-fuzz/issues/9583

File: infra/base-images/base-builder/compile_centipede [hunk 1]
```diff
@@ -25,11 +25,10 @@ fi
 
 cp "$BIN_DIR/libcentipede_runner.pic.a" "$LIB_FUZZING_ENGINE"
 
-export DFTRACING_FLAGS='-fsanitize-coverage=trace-loads'
 export CENTIPEDE_FLAGS=`cat "$SRC/centipede/clang-flags.txt" | tr '\n' ' '`
-export LIBRARIES_FLAGS="-Wno-unused-command-line-argument -ldl -lrt -lpthread $SRC/centipede/weak.o"
+export LIBRARIES_FLAGS="-Wno-unused-command-line-argument -Wl,-ldl -Wl,-lrt -Wl,-lpthread -Wl,$SRC/centipede/weak.o"
```
Comment (Alan32Liu) [inline infra/base-images/base-builder/compile_centipede:29]: nit: `-Wl` can [take a comma-separated list of flags](https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-wl-arg-arg2), we don't have to repeat it.

**Review by Alan32Liu — APPROVED** [review]

File: infra/base-images/base-builder/compile_centipede (hunk 1 above)
Comment (jonathanmetzman) [reply infra/base-images/base-builder/compile_centipede:29]: Good point. I'm gonna wait for this trial build to finish though.

**Comment by jonathanmetzman** [comment]
Pretty sure this works.

**Pull request merged**